apps/studio/src/pages/home/wallet/Portfolio.tsx

```diff
@@ -55,7 +65,40 @@ const Portfolio: FunctionComponent = () => {
 
     setSkeletonRows(rowsToRender);
     setRowsPerPage(rowsToRender);
-  }, [windowHeight]);
+
+    // TODO: Creates temporary earnings in place of backend table query
+    const walletSongsEarnings = songs.map((song) =>
+      createTempSongEarningsQueryData(song)
+    );
+
+    const combinedEarnings = walletSongsEarnings
+      .map((songEarning) => {
+        const totalRoyaltyAmount = songEarning.earningsData
+          // filter out earnings that are not within the filter period
+          .filter((songEarning) =>
+            isWithinFilterPeriod(
+              songEarning.createdAt,
+              walletPortfolioTableFilter
+            )
+          )
+          // combine earnings within the filter period
+          .reduce((acc, curr) => acc + curr.amount, 0);
+        return {
+          song: songEarning.song,
+          totalRoyaltyAmount,
+        };
+      })
+      // Sort filtered songs by descending totalRoyaltyAmount
+      .sort((a, b) => b?.totalRoyaltyAmount - a?.totalRoyaltyAmount)
+      // handle pagination for song earnings
+      .slice(pageIdx * skeletonRows, pageIdx * skeletonRows + rowsToRender);
+
+    setWalletSongsRoyaltyCombined(combinedEarnings);
+  }, [pageIdx, skeletonRows, songs, walletPortfolioTableFilter, windowHeight]);
```

Comment:
Yes, that would be the ideal with the endpoint, it would probably require us to feed the backend with an array of songs that are provided from the wallet. Otherwise this logic will still need to be consolidated on the frontend to deal with a single song earnings query, for summation of earnings, creation date filtering, and pagination. I will bring it up to the backend team as they start getting into building out the query.